lib/m2ts/caption-stream.js

```diff
@@ -256,30 +252,25 @@ CaptionStream.prototype.flush = function() {
 CaptionStream.prototype.reset = function() {
   this.latestDts_ = null;
   this.ignoreNextEqualDts_ = false;
-  this.activeCea608Channel_ = null;
-  // Since we don't know which channel is active until we get a control
-  // code that sets it, we start off with CEA608 handlers that just drop
-  // all the packets.
-  this.activeCea608Streams_ = [
-    packetDropper,
-    packetDropper
-  ];
+  this.activeCea608Channel_ = [null, null];
   this.ccStreams_.forEach(function(ccStream) {
     ccStream.reset();
   });
 };
 
 CaptionStream.prototype.dispatchCea608Packet = function(packet) {
-  if (this.setsChannel1Active(packet) && this.activeCea608Channel_ !== 1) {
-    this.activeCea608Channel_ = 1;
-    this.activeCea608Streams_ = [this.ccStreams_[0], this.ccStreams_[2]]; // CC1, CC3
-  } else if (this.setsChannel2Active(packet) && this.activeCea608Channel_ !== 2) {
-    this.activeCea608Channel_ = 2;
-    this.activeCea608Streams_ = [this.ccStreams_[1], this.ccStreams_[3]]; // CC2, CC4
+  // NOTE: packet.type is the CEA608 field
+  if (this.setsChannel1Active(packet) && this.activeCea608Channel_[packet.type] !== 0) {
```

Comment:
is the addition of ` && this.activeCea608Channel_[packet.type] !== 0` still necessary here? It doesn't seem like there would be any harm in setting `this.activeCea608Channel_[packet.type] = 0;` if it already is 0 (same for the check below)

Reply:
Yup, I suppose that's true.